src/components/SearchForm/AutocompleteCodeField.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { ReactElement } from 'react';
+import { Autocomplete, TextField } from '@mui/material';
+
+export interface AutocompleteCodeValue {
+  display: string;
+}
+
+export interface AutocompleteCodeFieldProps<T extends AutocompleteCodeValue> {
+  initialValue: T;
+  codeLoader: () => Promise<T[]>;
+  required?: boolean;
+  label?: string;
+}
+
+export interface AutocompleteCodeFieldState<T extends AutocompleteCodeValue> {
+  open: boolean;
+  loading: boolean;
+  options: T[];
+  value: T;
+}
+
+export class AutocompleteCodeField<T extends AutocompleteCodeValue> extends React.Component<
```

Comment:
The problem with this are the two member variables, loaded and optionLoadPromise. I suppose they could be moved to be state values? The problem I was having was that essentially every single time you opened the autocomplete, it would start an entirely new load.